packages/shared/lib/services/activity/activity.service.ts

```diff
@@ -205,6 +222,36 @@ export async function getTopLevelLogByEnvironment(environment_id: number, limit
         .offset(offset)
         .limit(limit);
 
+    if (status === 'success' || status === 'failure') {
+        logs.where({ success: status === 'success' });
+    }
+
+    if (status === 'in_progress') {
+        logs.where({ success: null });
+    }
+
+    if (script) {
+        logs.where({ operation_name: script });
+    }
+
+    if (connection) {
+        logs.where({ connection_id: connection });
+    }
+
+    if (integration) {
+        logs.where({ provider_config_key: integration });
+    }
+
+    if (date) {
+        const dateObj = new Date(date);
+        const month = (dateObj.getUTCMonth() + 1).toString().padStart(2, '0'); // Add leading zero if needed
+        const day = dateObj.getUTCDate().toString().padStart(2, '0'); // Add leading zero if needed
+        const formattedDate = `${month}/${day}/${dateObj.getUTCFullYear()}`;
+        logs.whereRaw(`to_char(to_timestamp(_nango_activity_logs.timestamp / 1000), 'MM/DD/YYYY') = '${formattedDate}'`);
```

Comment:
can we use something like this`date_trunc('day', timestamp) == ${{new Date(date.toDateString())}}`

Reply:
Updated to use
```
logs.whereRaw("date_trunc('day', to_timestamp(_nango_activity_logs.timestamp / 1000)) = ?", [formattedDate]);
```